nbextensions/appCell2/widgets/appCellWidget-fsm.js

```diff
@@ -314,55 +294,22 @@ define([], () => {
                 },
             },
             next: [
-                {
-                    mode: 'editing',
-                    params: 'incomplete',
-                },
-                {
-                    mode: 'editing-batch',
-                    params: 'incomplete',
-                },
-                {
-                    mode: 'editing-batch',
-                    params: 'complete',
-                    code: 'built',
-                },
-                {
-                    mode: 'execute-requested',
-                },
-                {
-                    mode: 'processing',
-                    stage: 'launched',
-                },
-                {
-                    mode: 'processing',
-                    stage: 'queued',
-                },
-                {
-                    mode: 'processing',
-                    stage: 'running',
-                },
-                {
-                    mode: 'success',
-                },
-                {
-                    mode: 'error',
-                    stage: 'launching',
-                },
-                {
-                    mode: 'error',
-                    stage: 'runtime',
-                },
-                {
-                    mode: 'internal-error',
-                },
+                STATE.EDITING_INCOMPLETE,
+                STATE.EDITING_BATCH_INCOMPLETE,
+                STATE.EDITING_BATCH_COMPLETE,
+                STATE.EXECUTE_REQUESTED,
+                STATE.PROCESSING_LAUNCHED,
+                STATE.PROCESSING_QUEUED,
+                STATE.PROCESSING_RUNNING,
+                STATE.COMPLETED,
+                STATE.LAUNCH_ERROR,
+                STATE.RUNTIME_ERROR,
+                STATE.INTERNAL_ERROR,
```

Comment:
not sure these are valid next states as you need to go through the `execute-requested` and `processing-launched` states to get the new job ID, but I'm not going to be changing them without thorough tests in place!